### PR TITLE
Added encoding supplemental doc

### PR DESCRIPTION
**Pull request by ddimatos**
##### SUMMARY
Modules `zos_fetch` and `zos_encode` both use `iconv` utility included in IBM Unix System Services (USS) and the supported encodings are over 600 which does not make this a good candidate to embed in the Ansible doc so an external static page and supported charsets has been created and linked back to the modules as well as the navigation tree. 

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
- `zos_fetch`
- `zos_encode` 
- web docs

### CODE REVIEW

**Comment by blakeinate** [comment]
Exception: ansible-test detected errors! 
 ERROR: Found 1 rstcheck issue(s) which need to be resolved:
ERROR: docs/source/supplementary.rst:32:0: Inline literal start-string without end-string.
ERROR: The 1 sanity test(s) listed below (out of 41) failed. See error output above for details.

**Comment by blakeinate** [comment]
Exception: ansible-test detected errors! 
 ERROR: Found 1 rstcheck issue(s) which need to be resolved:
ERROR: docs/source/supplementary.rst:32:0: Inline literal start-string without end-string.
ERROR: The 1 sanity test(s) listed below (out of 41) failed. See error output above for details.

**Comment by blakeinate** [comment]
Exception: ansible-test detected errors! 
 ERROR: Found 1 rstcheck issue(s) which need to be resolved:
ERROR: docs/source/supplementary.rst:23:0: Unexpected indentation.
ERROR: The 1 sanity test(s) listed below (out of 41) failed. See error output above for details.

**Comment by blakeinate** [comment]
Exception: ansible-test detected errors! 
 ERROR: Found 1 rstcheck issue(s) which need to be resolved:
ERROR: docs/source/supplementary.rst:52:0: Explicit markup ends without a blank line; unexpected unindent.
ERROR: The 1 sanity test(s) listed below (out of 41) failed. See error output above for details.

File: docs/source/supplementary.rst [hunk 1]
```diff
@@ -0,0 +1,1248 @@
...
+
+Many of the z/OS modules offer options to convert the data encodings that are
+read, copied, and transferred from a USS file or a data set. Encoding is managed
+by setting the source and destination character set options within the modules.
+
+A character set is an element of internationalization that maps and translates
+an alphabet or characters that are used in a particular language. A character
+set is made up of a series of code points or the numeric representation of a
+character. For example, the code point for the letter A in international EBCDIC
+is 0xC1.
```
Comment (RadhaV2020) [inline docs/source/supplementary.rst:19]: The definition is incomplete and im not in favor of using 'translating'. A
charset is not translating but just defines how an alphabet or character is
represented in a particular language. Please consider rewriting. IBM terminology
defines it differently
Suggested rewrite: A character set is an element of internationalization that
describes/ defines how an alphabet or character is represented in a particular
language.
IBM approved terminology describes it as :



noun
A defined set of characters with no coded representation assumed that can be recognized by a configured hardware or software system. A character set can be defined by alphabet, language, script, or any combination of these items.

Source: AIX Glossary, C/C++ for Workstations, C/C++ for z/OS, Content Manager OnDemand, DB2 Glossary, DB2 for z/OS, Dictionary of Printing, Globalization Leadership Team, IBM i, Language Environment, Rational Developer for IBM I, Terminology Extraction Process, Translation Services Center, z/OS UNIX System Services
Subject area: Information technology // Software
A set of binary codes that represent specific text characters.

Source: Domino, Notes, Terminology Extraction Process, Translation Services Center
Subject area: Information technology // Software
A list of characters (letters, numbers, and symbols such as #, $, and &) that are recognized by computer hardware and software.

Source: Sterling B2B Integrator
Subject area: Information technology // Software

source: http://tlwi.w3-969.ibm.com/standards/terminology/cgi-bin/lookup?ug=corporate&term=character+set&submit=Search&source=en&target_switch=none&template=simple&db_name=LOGOS&11=main+form&11=acronym%7Eabbreviation&11=prohibited

Comment (ddimatos) [reply docs/source/supplementary.rst:19]: I am ok with the suggested change, thank you.

**Comment by blakeinate** [comment]
Exception: One or more pytest testcases failed! 
 ============================= test session starts ==============================
platform linux -- Python 3.7.6, pytest-5.3.0, py-1.8.1, pluggy-0.13.1
ansible: 2.9.6
rootdir: /var/lib/jenkins/workspace/zos-ansible-2/ibm_zos_core/tests
plugins: ansible-2.2.2, mock-1.12.1
collected 251 items

functional/module_utils/test_arg_parser.py ............................. [ 11%]
..............................................................           [ 36%]
functional/modules/test_zos_data_set_func.py ........................... [ 47%]
...................F.....                                                [ 56%]
functional/modules/test_zos_encode_func.py .....................         [ 65%]
functional/modules/test_zos_fetch_func.py ........................       [ 74%]
functional/modules/test_zos_job_output_func.py ..                        [ 75%]
functional/modules/test_zos_job_query_func.py .                          [ 76%]
functional/modules/test_zos_job_submit_func.py ...                       [ 77%]
functional/modules/test_zos_operator_action_query_func.py .............. [ 82%]
............                                                             [ 87%]
functional/modules/test_zos_operator_func.py ......                      [ 90%]
functional/modules/test_zos_ping_func.py .                               [ 90%]
functional/modules/test_zos_tso_command_func.py ..........               [ 94%]
unit/test_zos_operator_action_query_unit.py .........                    [ 98%]
unit/test_zos_operator_unit.py .....                                     [100%]

=================================== FAILURES ===================================
___________________ test_data_set_deletion_when_present[lds] ___________________

ansible_zos_module = <pytest_ansible.host_manager.v28.HostManagerV28 object at 0x7f37ddd0ea10>
dstype = 'lds'

    @pytest.mark.parametrize("dstype", data_set_types)
    def test_data_set_deletion_when_present(ansible_zos_module, dstype):
        hosts = ansible_zos_module
        hosts.all.zos_data_set(name="imstestl.ims1.test05", state="present", type=dstype)
        results = hosts.all.zos_data_set(name="imstestl.ims1.test05", state="absent")
        for result in results.contacted.values():
>           assert result.get("changed") is True
E           AssertionError: assert False is True
E            +  where False = <built-in method get of dict object at 0x7f37dd442370>('changed')
E            +    where <built-in method get of dict object at 0x7f37dd442370> = {'_ansible_no_log': False, 'changed': False, 'invocation': {'module_args': {'batch': None, 'data_class': None, 'format': None, 'name': 'imstestl.ims1.test05', ...}}, 'message': ''}.get

functional/modules/test_zos_data_set_func.py:308: AssertionError
---------------------------- Captured stdout setup -----------------------------
setting up inventory plugins
Parsed ec33025a.vmec.svl.ibm.com, inventory source with host_list plugin
================== 1 failed, 250 passed in 2056.81s (0:34:16) ==================

**Comment by blakeinate** [comment]
Exception: One or more pytest testcases failed! 
 ============================= test session starts ==============================
platform linux -- Python 3.7.6, pytest-5.3.0, py-1.8.1, pluggy-0.13.1
ansible: 2.9.6
rootdir: /var/lib/jenkins/workspace/zos-ansible-2/ibm_zos_core/tests
plugins: ansible-2.2.2, mock-1.12.1
collected 251 items

functional/module_utils/test_arg_parser.py ............................. [ 11%]
..............................................................           [ 36%]
functional/modules/test_zos_data_set_func.py F.......................... [ 47%]
.........................                                                [ 56%]
functional/modules/test_zos_encode_func.py .....................         [ 65%]
functional/modules/test_zos_fetch_func.py ........................       [ 74%]
functional/modules/test_zos_job_output_func.py ..                        [ 75%]
functional/modules/test_zos_job_query_func.py .                          [ 76%]
functional/modules/test_zos_job_submit_func.py ...                       [ 77%]
functional/modules/test_zos_operator_action_query_func.py .............. [ 82%]
............                                                             [ 87%]
functional/modules/test_zos_operator_func.py ......                      [ 90%]
functional/modules/test_zos_ping_func.py .                               [ 90%]
functional/modules/test_zos_tso_command_func.py ..........               [ 94%]
unit/test_zos_operator_action_query_unit.py .........                    [ 98%]
unit/test_zos_operator_unit.py .....                                     [100%]

=================================== FAILURES ===================================
_ test_data_set_catalog_and_uncatalog[\n//CREPDS    JOB (T043JM,JM00,1,0,0,0),'CREATE PDS',CLASS=R,\n//             MSGCLASS=X,MSGLEVEL=1,NOTIFY=OMVSADM\n//STEP1    EXEC PGM=IDCAMS\n//SYSPRINT DD  SYSOUT=A\n//SYSIN    DD   *\n     ALLOC -\n           DSNAME('USER.PRIVATE.TESTDS') -\n           NEW -\n           VOL(000000) -\n           DSNTYPE(PDS)\n/*\n] _

ansible_zos_module = <pytest_ansible.host_manager.v28.HostManagerV28 object at 0x7f7b8ec5ead0>
jcl = "\n//CREPDS    JOB (T043JM,JM00,1,0,0,0),'CREATE PDS',CLASS=R,\n//             MSGCLASS=X,MSGLEVEL=1,NOTIFY=OMVSADM\n/...n           DSNAME('USER.PRIVATE.TESTDS') -\n           NEW -\n           VOL(000000) -\n           DSNTYPE(PDS)\n/*\n"

    @pytest.mark.parametrize(
        "jcl",
        [PDS_CREATE_JCL, KSDS_CREATE_JCL, RRDS_CREATE_JCL, ESDS_CREATE_JCL, LDS_CREATE_JCL],
    )
    def test_data_set_catalog_and_uncatalog(ansible_zos_module, jcl):
        hosts = ansible_zos_module
        hosts.all.zos_data_set(
            name=DEFAULT_DATA_SET_NAME, state="cataloged", volume=DEFAULT_VOLUME
        )
        hosts.all.zos_data_set(name=DEFAULT_DATA_SET_NAME, state="absent")
        hosts.all.file(path=TEMP_PATH, state="directory")
        hosts.all.shell(cmd="echo {0} > {1}/SAMPLE".format(quote(jcl), TEMP_PATH))
        results = hosts.all.zos_job_submit(
            src=TEMP_PATH + "/SAMPLE", location="USS", wait=True
        )
        hosts.all.file(path=TEMP_PATH, state="absent")
        # verify data set creation was successful
        for result in results.contacted.values():
            assert result.get("jobs")[0].get("ret_code").get("msg_code") == "0000"
        # verify first uncatalog was performed
        results = hosts.all.zos_data_set(name=DEFAULT_DATA_SET_NAME, state="uncataloged")
        for result in results.contacted.values():
            assert result.get("changed") is True
        # verify second uncatalog shows uncatalog already performed
        results = hosts.all.zos_data_set(name=DEFAULT_DATA_SET_NAME, state="uncataloged")
        for result in results.contacted.values():
            assert result.get("changed") is False
        # recatalog the data set
        results = hosts.all.zos_data_set(
            name=DEFAULT_DATA_SET_NAME, state="cataloged", volume=DEFAULT_VOLUME
        )
        for result in results.contacted.values():
>           assert result.get("changed") is True
E           assert False is True
E            +  where False = <built-in method get of dict object at 0x7f7b8d6412d0>('changed')
E            +    where <built-in method get of dict object at 0x7f7b8d6412d0> = {'_ansible_no_log': False, 'changed': False, 'exception': '  File "/tmp/ansible_zos_data_set_payload_hlttanjd/ansible_...cation': {'module_args': {'batch': None, 'data_class': None, 'format': None, 'name': 'USER.PRIVATE.TESTDS', ...}}, ...}.get

functional/modules/test_zos_data_set_func.py:143: AssertionError
================== 1 failed, 250 passed in 1719.33s (0:28:39) ==================

**Comment by blakeinate** [comment]
Exception: One or more pytest testcases failed! 
 ============================= test session starts ==============================
platform linux -- Python 3.7.6, pytest-5.3.0, py-1.8.1, pluggy-0.13.1
ansible: 2.9.6
rootdir: /var/lib/jenkins/workspace/zos-ansible-2/ibm_zos_core/tests
plugins: ansible-2.2.2, mock-1.12.1
collected 251 items

functional/module_utils/test_arg_parser.py ............................. [ 11%]
..............................................................           [ 36%]
functional/modules/test_zos_data_set_func.py F.......................... [ 47%]
.........................                                                [ 56%]
functional/modules/test_zos_encode_func.py .....................         [ 65%]
functional/modules/test_zos_fetch_func.py ........................       [ 74%]
functional/modules/test_zos_job_output_func.py ..                        [ 75%]
functional/modules/test_zos_job_query_func.py .                          [ 76%]
functional/modules/test_zos_job_submit_func.py ...                       [ 77%]
functional/modules/test_zos_operator_action_query_func.py .............. [ 82%]
............                                                             [ 87%]
functional/modules/test_zos_operator_func.py ......                      [ 90%]
functional/modules/test_zos_ping_func.py .                               [ 90%]
functional/modules/test_zos_tso_command_func.py ..........               [ 94%]
unit/test_zos_operator_action_query_unit.py .........                    [ 98%]
unit/test_zos_operator_unit.py .....                                     [100%]

=================================== FAILURES ===================================
_ test_data_set_catalog_and_uncatalog[\n//CREPDS    JOB (T043JM,JM00,1,0,0,0),'CREATE PDS',CLASS=R,\n//             MSGCLASS=X,MSGLEVEL=1,NOTIFY=OMVSADM\n//STEP1    EXEC PGM=IDCAMS\n//SYSPRINT DD  SYSOUT=A\n//SYSIN    DD   *\n     ALLOC -\n           DSNAME('USER.PRIVATE.TESTDS') -\n           NEW -\n           VOL(000000) -\n           DSNTYPE(PDS)\n/*\n] _

ansible_zos_module = <pytest_ansible.host_manager.v28.HostManagerV28 object at 0x7f20f75abd90>
jcl = "\n//CREPDS    JOB (T043JM,JM00,1,0,0,0),'CREATE PDS',CLASS=R,\n//             MSGCLASS=X,MSGLEVEL=1,NOTIFY=OMVSADM\n/...n           DSNAME('USER.PRIVATE.TESTDS') -\n           NEW -\n           VOL(000000) -\n           DSNTYPE(PDS)\n/*\n"

    @pytest.mark.parametrize(
        "jcl",
        [PDS_CREATE_JCL, KSDS_CREATE_JCL, RRDS_CREATE_JCL, ESDS_CREATE_JCL, LDS_CREATE_JCL],
    )
    def test_data_set_catalog_and_uncatalog(ansible_zos_module, jcl):
        hosts = ansible_zos_module
        hosts.all.zos_data_set(
            name=DEFAULT_DATA_SET_NAME, state="cataloged", volume=DEFAULT_VOLUME
        )
        hosts.all.zos_data_set(name=DEFAULT_DATA_SET_NAME, state="absent")
        hosts.all.file(path=TEMP_PATH, state="directory")
        hosts.all.shell(cmd="echo {0} > {1}/SAMPLE".format(quote(jcl), TEMP_PATH))
        results = hosts.all.zos_job_submit(
            src=TEMP_PATH + "/SAMPLE", location="USS", wait=True
        )
        hosts.all.file(path=TEMP_PATH, state="absent")
        # verify data set creation was successful
        for result in results.contacted.values():
            assert result.get("jobs")[0].get("ret_code").get("msg_code") == "0000"
        # verify first uncatalog was performed
        results = hosts.all.zos_data_set(name=DEFAULT_DATA_SET_NAME, state="uncataloged")
        for result in results.contacted.values():
            assert result.get("changed") is True
        # verify second uncatalog shows uncatalog already performed
        results = hosts.all.zos_data_set(name=DEFAULT_DATA_SET_NAME, state="uncataloged")
        for result in results.contacted.values():
            assert result.get("changed") is False
        # recatalog the data set
        results = hosts.all.zos_data_set(
            name=DEFAULT_DATA_SET_NAME, state="cataloged", volume=DEFAULT_VOLUME
        )
        for result in results.contacted.values():
>           assert result.get("changed") is True
E           assert False is True
E            +  where False = <built-in method get of dict object at 0x7f20f5c92e10>('changed')
E            +    where <built-in method get of dict object at 0x7f20f5c92e10> = {'_ansible_no_log': False, 'changed': False, 'exception': '  File "/tmp/ansible_zos_data_set_payload_jarcoxds/ansible_...cation': {'module_args': {'batch': None, 'data_class': None, 'format': None, 'name': 'USER.PRIVATE.TESTDS', ...}}, ...}.get

functional/modules/test_zos_data_set_func.py:143: AssertionError
================== 1 failed, 250 passed in 1755.35s (0:29:15) ==================

**Pull request merged**